README.md

```diff
@@ -1,222 +1,515 @@
+
+#### Using Variables Before Declaration
+
+Variables must be declared before they are used. Hoisting makes it possible to accidentally use a variable before it's declared.
+
+#### Aligning Assignments
+
+Variable assignments shouldn't be aligned by value.
 
-Variable assignments should not be aligned by value.
 **Don't do this:**
+
 ```javascript
-var foo      = 'foo';
-var bar      = 'bar';
-var fizzbuzz = 'fizzbuzz';
+let foo      = 'foo';
+let bar      = 'bar';
+let fizzbuzz = 'fizzbuzz';
 
-var obj = {
+let obj = {
   foo     : 'foo',
   bar     : 'bar',
   fizzbuzz: 'fizzbuzz'
 };
 ```
+
 **Do this instead:**
+
 ```javascript
-var foo = 'foo';
-var bar = 'bar';
-var fizzbuzz = 'fizzbuzz';
+let foo = 'foo';
+let bar = 'bar';
+let fizzbuzz = 'fizzbuzz';
 
-var obj = {
+let obj = {
   foo: 'foo',
   bar: 'bar',
   fizzbuzz: 'fizzbuzz'
 };
 ```
 
-## Anonymous Callback Functions
+### Constants
+
+This section conflicts with the Google JavaScript Style Guide.
+The following rules should be followed instead.
 
-Prefer `var`ed function expressions over anonymous functions for callbacks where it makes sense.
+* The `@const` tag shouldn't be used.
+* When available, the `const` keyword should be used.
+* Consider [freezing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze) object constants when applicable.
+
+### Functions
+
+Function expressions should be used instead of function declarations.
 
 **Don't do this:**
+
+```javascript
+function foo() {
+  /// [...]
+};
+```
+
+**Do this instead:**
+
+```javascript
+let foo = function() {
+  /// [...]
+};
+```
+
+#### Anonymous Functions
+
+Function expressions should be used instead of anonymous functions where applicable.
+[IIFEs](https://en.wikipedia.org/wiki/Immediately-invoked_function_expression) shouldn't be used.
+
+**Don't do this:**
+
 ```javascript
 $('.someElement').click(function(){
   console.log('It was clicked!');
 });
 ```
+
+```javascript
+fetchFoo()
+  .then(function(foo) {
+    // Do something
+  });
+```
+
 **Do this instead:**
+
 ```javascript
-var someElementClicked = function(){
+const someElementClicked = function(){
   console.log('It was clicked!');
 });
+
 $('.someElement').click(someElementClicked);
 ```
-This is especially important in `new`ed objects like Backbone Models and Views, where scope is important, and callbacks often reference other properties on the instance.
 
-## Manipulating `this` context
+```javascript
+const calculateBar = function(foo) {
+  // Do something
+};
+
+fetchFoo()
+  .then(calculateBar);
+```
+
+### Modules
 
-Prefer calling functions in the required scope vs. saving `this` in a `that` or `self` variable.
+Modules should be small, [composable](http://en.wikipedia.org/wiki/Composability), [stateless](http://en.wikipedia.org/wiki/State_(computer_science)), and self contained.
+
+#### Exporting
+
+Functions and primitive types shouldn't be default exports.
```

Comment:
Add an exception here: internal functions CAN be exported to a parent module iff (double 'f' intentional) they are entirely encapsulated by the parent.

```js
import a from 'func-a';
import b from 'func-b';

exports default {
  a: a,
  b: b
};
```

We care about the public API/interface of a module obeying this rule, not the internal implementation.